src/server/game/Spells/Auras/SpellAuraEffects.cpp

```diff
@@ -3605,6 +3605,26 @@ void AuraEffect::HandleOverrideAttackPowerBySpellPower(AuraApplication const* au
     target->UpdateAttackPowerAndDamage(true);
 }
 
+void AuraEffect::HandleModVersatilityByPct(AuraApplication const* aurApp, uint8 mode, bool apply) const
+{
+    if (!(mode & (AURA_EFFECT_HANDLE_CHANGE_AMOUNT_MASK | AURA_EFFECT_HANDLE_STAT)))
+        return;
+
+    Unit* target = aurApp->GetTarget();
```

Comment:
`Player* target = aurApp->GetTarget()->ToPlayer()`
and all other ToPlayers are gone from the function